src/session.ts

```diff
@@ -0,0 +1,272 @@
+      $items`grogtini, bodyslam, dirty martini, vesper, cherry bomb, sangria del diablo`
+    ),
+    ...manyToOne(
+      $item`earthenware muffin tin`,
+      $items`blueberry muffin, bran muffin, chocolate chip muffin`
+    ),
+  ]);
+
+  const inventory = new Map<Item, number>();
+  for (const [itemStr, quantity] of Object.entries(mySessionItems())) {
+    const item = toItem(itemStr);
+    const mappedItem = itemMappings.get(item) ?? item;
+    inventory.set(mappedItem, quantity + (inventory.get(mappedItem) ?? 0));
+  }
+  return inventory;
+}
+
+/**
+ * Performa a binary element-wise operation on two inventories
+ * @param a The LHS inventory to perform the operation on
+ * @param b The RHS inventory to perform the operation on
+ * @param op an operator to compute between the sets
+ * @param commutative if true use the value of b for any items not in a. if false, ignore values not in a
+ * @returns a new map representing the combined inventories
+ */
+function inventoryOperation(
+  a: Map<Item, number>,
+  b: Map<Item, number>,
+  op: (aPart: number, bPart: number) => number,
+  commutative: boolean
+): Map<Item, number> {
+  // return every entry that is in a and not in b
+  const difference = new Map<Item, number>();
+
+  for (const [item, quantity] of a.entries()) {
+    const combinedQuantity = op(quantity, b.get(item) ?? 0);
+    difference.set(item, combinedQuantity);
+  }
+  if (commutative) {
+    for (const [item, quantity] of b.entries()) {
+      if (!a.has(item)) {
+        difference.set(item, quantity);
+      }
+    }
+  }
+  const diffEntries: [Item, number][] = [...difference.entries()];
+
+  return new Map<Item, number>(diffEntries.filter((value) => value[1] !== 0));
+}
+
+/**
+ * An entry showing the value of each Item in a session
+ * @member item the item associated with this detail
+ * @member value the numeric value of the full quantity of items (to get value of each item, do value / quantity) (can be negative)
+ * @member quantity the number of items for this detail
+ */
+interface ItemDetail {
+  item: Item;
+  value: number;
+  quantity: number;
+}
+
+/**
+ * The full value (in meat) results of a session
+ * @member meat the value of this session in pure meat
+ * @member items the value of the items in this session in meat
+ * @member total sum of meat and items
+ * @member itemDetails a list of the detailed accounting for each item in this session
+ */
+interface ItemResult {
+  meat: number;
+  items: number;
+  total: number;
+  itemDetails: ItemDetail[];
+}
+
+/**
+ * A wrapper around tracking items and meat gained from this session
+ * Smartly handles foldables being added/removed based on their state
+ * Provides operations to add sessions and subtract Sessions so you can isolate the value of each Session using a baseline
+ * @member meat the raw meat associated with this Session
+ * @member items a map representing the items gained/lost during this Session
+ */
+export class Session {
```

Comment:
```suggestion
export default class Session {
```

Reply:
i don't think we are using default exports anywhere else in libram

Reply:
When I first wrote `Dungeon.ts`, gausie suggested that I export `Dungeon` as default, that's what I was basing this on